src/PhpSpreadsheet/Reader/Html.php

```diff
@@ -147,10 +147,9 @@ public function canRead(string $filename): bool
             return false;
         }
 
-        $beginning = $this->readBeginning();
-        if (preg_match(self::STARTS_WITH_BOM, $beginning)) {
-            return true;
-        }
+        $beginning = preg_replace(self::STARTS_WITH_BOM, '', $this->readBeginning());
+        assert($beginning !== null);
```

Comment:
We can't have an assert in the code. However, you can achieve the same affect by adding a null coercion to the previous statement.
```php
$beginning = preg_replace(...) ?? '';
```
Assuming that all the tests still work, and they should, I will be able to merge your PR once this change is made.